cmd/odo/common_test.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"os"
+	"testing"
+
+	"github.com/redhat-developer/odo/pkg/config"
+	envcontext "github.com/redhat-developer/odo/pkg/config/context"
+	"github.com/redhat-developer/odo/pkg/odo/cli"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	"github.com/spf13/pflag"
+	"k8s.io/klog"
+)
+
+func resetGlobalFlags() {
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	pflag.CommandLine = pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
+	klog.InitFlags(nil)
+}
+
+func runCommand(
+	t *testing.T,
+	args []string,
+	clientset clientset.Clientset,
+	f func(err error, stdout, stderr string),
+) {
+
+	// We are running the test on a new and empty directory (on real filesystem)
+	originWd, err := os.Getwd()
+	if err != nil {
+		t.Error(err)
+	}
+	defer func() {
+		_ = os.Chdir(originWd)
+	}()
+	cwd := t.TempDir()
+	err = os.Chdir(cwd)
+	if err != nil {
+		t.Error(err)
```

Comment:
Same point about stopping at this point, since `t.Error` would continue the execution.